### DISCUSSION TITLE
Windows XP Compatible v2.7 build here

### DISCUSSION THREAD

**Discussion by Gerwin2k** [LostArtefacts/TRX#527] [Show and tell]
Thanks for making this great new version Tomb Raider 1. I only found out about it recently with v2.7.
Because the Tomb1main v2.x series dropped several dependencies, a back-port may be a lot easier then before. Indeed it was a straightforward matter, and for anyone who cares for it, here is a v2.7 than can run on Windows XP: 

See http://www.gb-homepage.nl/index.htm  and select the "Downloads" section. Edit: I It is now in a "Back-ports" section.

The package comes with required FFmpeg dll's, and those are 40MB unfortunately. But I don't feel like building them myself just to shave off the unused video codecs.

**Comment by rr-** [comment]
Nice work. Thanks.

**Reply by Gerwin2k** [reply]
Congrats with the new v2.11 release. 
I added an updated v2.11 Windows XP build to the above mentioned website a minute ago.

**Comment by Gerwin2k** [comment]
There is an v2.15 update of the XP-mod package now. Link as above.
Some months ago, I build custom FFmpeg libraries. With these, the package became a lot smaller.

**Reply by rr-** [reply]
We'll be happy to reduce the bundle size and we're open to patches to our dockerfiles.

**Reply by Gerwin2k** [reply]
I meant: My earlier XP-mod packages with separate DLL files were quite large. Full FFmpeg libraries being around 40MB. I improved that, and it is now a 1MB executable with 6MB of DLL files.

The normal Tomb1main releases here on github are fine in size. Being a statically linked executable of 6MB. So FFmpeg is already slimmed down, obviously.

**Reply by rr-** [reply]
Thanks. This makes sense.

**Comment by Gerwin2k** [comment]
Earlier I added some updated versions for Tomb Raider 1 on Windows XP.
TR1X v4.0.1 
TR1X v4.2 

A minute ago I added:
TR2X v0.1.0
I had to update to FFMpeg 5 for this one. 

Congrats on the Tomb Raider 2 work. Much appreciated!

**Comment by Gerwin2k** [comment]
Recently I build TR1X v4.13.2 and TR2X v1.3.2. :)

It did not go easy. I cannot use meson and convert the meson to Code::Blocks projects manually, then use MinGW 10.3.0. So I tried to do it properly and make the libtrx library first. then build the games and link them to libtrx. I get no error messages, but the resulting games are funky. 

Eventually I settled for another method: Put libtrx in the source tree of the TR1X / TR2X then build it like that in one go. There is one important thing though: the "game" folder of each game has to be renamed to something, for example to "gam1", then batch edit all references to these files accordingly. If I do not do this, then the compiler will find header files with the same name, like one of them is in the game, but a header with the same name is in libtrx. The compiler will use the wrong headers and the game will be funky again. With the renaming, none of these problems occur.

**Comment by Gerwin2k** [comment]
Added a TRX v1.7.1 Windows XP Build, with full TR3 support. 

The LibTRX complications mentioned above, no longer apply. So that is great. :)
